### PR TITLE
Title

### LINKED ISSUE #57: Edit title of a pin
As a user I would like to be able to name my pins

### PR DESCRIPTION
**Pull request by asu91**
- Center On Me button now pans smoothly to current location
- deleting pins will also delete from recent if possible
- can now edit pin titles from map and pin list
- pins removed in database will be removed on map in real-time

closes #57

### CODE REVIEW

**Comment by jake-shasteen** [comment]
@asu91 Reviewing

**Comment by bignell** [comment]
@asu91 And my axe!

File: app/components/PinEditButton.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const { pin, deletePin, hideButton } = this.props;
+    return(
+      <Button
+        style={[styles.bubble, styles.button]}
+
+        onPress={ () => {
+          hideButton();
+          AlertIOS.prompt(
+            pin.title,
+            'Editting Pin',
```
Comment (bignell) [inline app/components/PinEditButton.js]: We can probably refine the language of the pop-up (for one thing, "Editing" has one "t" :stuck_out_tongue_closed_eyes:), but that's a minor thing and doesn't interfere with the functionality of this PR.

**Comment by jake-shasteen** [comment]
On first run, I notice that the 'drop new pin' button is no longer present. Also, when I turn the device to landscape mode, the 'Center On Me' button disappears.

I notice that when I drop a pin it makes me enter a title. It would be good if that space were prefilled with something so that if I quickly just hit 'ok' it isn't blank. Some default placeholder.

I also observe that the testObj pins are not showing up on the map in the area of Hack Reactor. Is this expected behavior?

It seems as though the pin title changes persist over restarts, so they are being populated to the firebase server. Nice!

Anyway -- the styling makes 'drop new pin' kind of unusable right now, and landscape mode is right out.

I'll do a deep dive into the code after lunch.

**Comment by jake-shasteen** [comment]
Merged! Nice job with the polish on this one, it looks great. :heavy_check_mark:

**Pull request merged**